src/sbt-metering/metering-interface.ts

```diff
@@ -0,0 +1,58 @@
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { DetailType } from '../../utils';
+
+/**
+ * Optional interface that allows specifying both
+ * the function to trigger and the event that will trigger it.
+ */
+export interface IFunctionTrigger {
+  /**
+   * The function definition.
+   */
+  readonly handler: IFunction;
+
+  /**
+   * The detail-type that will trigger the handler function.
+   */
+  readonly trigger: DetailType;
+}
+
+/**
+ * Encapsulates the list of properties for an IMetering construct.
+ */
+export interface IMetering {
+  /**
+   * The function to trigger to create a new customer.
+   * (Customer in this context is a tenant.)
+   */
+  createCustomerFunction: IFunction | IFunctionTrigger;
```

Comment:
An AWS account building SaaS application will be an account in Amberflo.
A tenant is a client of the SaaS application and will be a customer in Amberflo.

Since tenant is a customer of the SaaS application, this function is named as createCustomerFunction. I have tried it to keep it similar to what they have in [`billing-interface`](https://github.com/awslabs/sbt-aws/blob/main/src/control-plane/billing/billing-interface.ts#L49)

If we feel `createTenantFunction` is a better choice we can use that.

Reply:
It is still not clear to me the answer to this question:
> in Amberflo terms, this function will create a "customer" in Amberflo or an "account" in Amberflo?

Reply:
This function will create a customer in Amberflo.